hw/top_earlgrey/cdc/cdc_waivers.spi_device.tcl

```diff
@@ -0,0 +1,14 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Verix CDC waiver file
+# Expression:
+#  ControlSignal==""
+#  ReconSignal==""
+#  MultiClockDomains=="IO_DIV2_CLK::IO_DIV4_CLK"
+
+set_rule_status -rule {W_RECON_GROUPS} -status {Waived}         \
+   -expression {(ControlSignal=~"*u_status_23_to_1_sync.*") &&       \
+                (ReconSignal=~"*u_spid_status.outclk_p2s_byte_*")} \
+   -comment {status bits are converged into SPI line. But transmit bit-by-bit. OK to send old and new status bits in a transaction}
```

Comment:
Hm... you may have identified a potential problem. I'm not sure how significant yet, but I'll outline the thinking:

The other bits are controlled by either "Write Enable" (which I'd suggest handling in hardware) and "Write Status". For Write Status, the host will expect that the bits will all be settled if it sees that BUSY is clear. The current timing may have BUSY update after other bits have sent, but in time for the cleared BUSY bit to be sent out.

If we think there is a potential for skew, we could maintain a "next_state" and a host-visible "committed_state," then update the committed_state only during the command phase. We'd also want to ensure that BUSY can't update ahead of the rest of the bits, but this doesn't necessarily need hardware: We can tell software to issue two separate transactions, for example.

Reply:
if you are concerned about old and new, why not synchronize the write pulse across and handle all the status updates in one shot once in sck? 
The other bits could be updated, and the busy could be cleared if the incoming value is 0.

Reply:
yeah.. that would be better I think. So, after SW updates, the logic blocks further updates from the SW and store the write data into registers and pulse sync to SPI clock domain. Then, clear the block when it actually sees the update @ SPI clock domain.

maybe using `prim_sync_reqack_data` would be better.

Reply:
however, what i'm still saying STILL has the problem with "what happens before the module ever runs a single transaction".
Are we okay with the status changing in the middle of a read? On the status read you might still need to latch the output once again at the end of the command phase probably.

Reply:
So, if we opted for the reqack sync (with data), then if SW updates the STATUS prior to the SPI transaction (when init() ?), then it should be OK as the data is applied to the SPI clock domain prior to the command receiving.

If we use reqack when the first SPI transaction is completed (CS# de-assertion), then... I don't know...

Anyway, I just realized that reqack and 2FF are same here unless there are staged/ commited steps as Alex mentioned. Transferring happens bit-by-bit. So, 2FF should not differ from reqack in terms of the error (weird) scenario that old Status is being sent

Reply:
Eunchan do you latch the return data when a read status command is received? So that any changes during returning do not get reflected? Because I honestly feel like that would be the best way.  You could update all the status in one shot at the same time, or not at all. 

The main thought I have about the 2FF is that the sync may happen on different cycles, so maybe in the worst case you hit RIGHT on the update and one bit gets picked up one cycle before/after the others.

Reply:
I think your approach is the best. Kind of similar to Alex's comment.

The thing related to 2FF is: As Read Status returns the data bit-by-bit, one clock diff should be OK. So if bit 3 updates first and bit 2 updates a clock later, new bit 3 and 2 are being transferred. If opposite, then old values are transferred. There will be no bit 3 (new) , bit 2 (old) case.

Reply:
So, the approach would be:

- Change to reqack to ensure all bits are changed at the same time
- Latch Status registers when receiving the Status command (or always at the 7th cycle) then returns the data.

Reply:
ah i see, thanks for clarifying.